tools/Dashboard/DevHome.Dashboard/ComSafeWidgetObjects/ComSafeWidgetDefinition.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using DevHome.Common.Extensions;
+using DevHome.Dashboard.Services;
+using Microsoft.UI.Xaml;
+using Microsoft.Windows.Widgets.Hosts;
+
+namespace DevHome.Dashboard.ComSafeWidgetObjects;
+
+/// <summary>
+/// Since WidgetDefinitions are OOP COM objects, we need to wrap them in a safe way to handle COM exceptions
+/// that arise when the underlying OOP object vanishes. All WidgetDefinitions should be wrapped in a
+/// ComSafeWidgetDefinition and calls to the WidgetDefinition should be done through the ComSafeWidgetDefinition.
+/// This class will handle the COM exceptions and get a new OOP WidgetDefinition if needed.
+/// All APIs on the IWidgetDefinition and IWidget2Definition interfaces are reflected here.
+/// </summary>
+public class ComSafeWidgetDefinition

```

Comment:
This is a not-correct idea.  Why not surround the class with a try/catch(ComException)?